service/history/queues/executable.go

```diff
@@ -163,15 +167,29 @@ func NewExecutable(
 	return executable
 }
 
-func (e *executableImpl) Execute() error {
+func (e *executableImpl) Execute() (retErr error) {
 	if e.State() == ctasks.TaskStateCancelled {
 		return nil
 	}
 
-	ctx := metrics.AddMetricsContext(context.Background())
-	namespace, _ := e.namespaceRegistry.GetNamespaceName(namespace.ID(e.GetNamespaceID()))
+	namespaceName, _ := e.namespaceRegistry.GetNamespaceName(namespace.ID(e.GetNamespaceID()))
```

Comment:
do you also need to handle this error?

Reply:
No I don't think so, if there's any error, `namespaceName` will be an empty string, which is a reasonable value for the rest of the logic.